lib/phoenix_live_view/helpers.ex

```diff
@@ -370,6 +370,90 @@ defmodule Phoenix.LiveView.Helpers do
     EEx.compile_string(expr, options)
   end
 
+  @doc """
+  Returns the entry errors for an upload.
+
+  The following errors may be returned:
+
+    * `:too_large` - The entry exceeded the `:max_file_size` constraint
+    * `:too_many_files` - The number of selected files exceeds the `:max_entries` constraint
+    * `:not_acceptable` - The entry does not match the `:accept` MIME types
+
+  ## Examples
+
+      def error_to_string(:too_large), do: "Too large"
+      def error_to_string(:too_many_files), do: "You have selected too many files"
+
+      <%= for entry <- @uploads.avatar.entries do %>
+        <%= for err <- upload_errors(@uploads.avatar, entry) %>
+          <div class="alert alert-danger">
+            <%= error_to_string(err) %>
+          </div>
+        <% end %>
+      <% end %>
+  """
+  def upload_errors(
+        %Phoenix.LiveView.UploadConfig{} = conf,
+        %Phoenix.LiveView.UploadEntry{} = entry
+      ) do
+    for {ref, error} <- conf.errors, ref == entry.ref, do: error
+  end
+
+  @doc """
+  Generates an image preview on the client for a selected file.
+
+  ## Examples
+
+      <%= for entry <- @uploads.avatar.entries do %>
+        <%= live_img_preview entry, width: 75 %>
+      <% end %>
+  """
+  def live_img_preview(%Phoenix.LiveView.UploadEntry{ref: ref} = entry, opts \\ []) do
+    binding_prefix = opts[:binding_prefix] || "phx-"
+
+    opts =
+      Keyword.merge(opts,
+        id: "phx-preview-#{ref}",
+        data_phx_upload_ref: entry.upload_ref,
+        data_phx_entry_ref: ref
+      ) ++ [{"#{binding_prefix}hook", "Phoenix.LiveImgPreview"}]
+
+    assigns = %{opts: opts}
+
+    ~L"""
+    <%= Phoenix.HTML.Tag.content_tag(:img, "", @opts) %>
+    """
+  end
+
+  @doc """
+  Builds a file input tag for a LiveView upload.
+
+  Options may be passed through to the tag builder for custom attributes.
+
+  ## Examples
+
+      <%= live_file_input @uploads.avatar %>
+  """
+  def live_file_input(%Phoenix.LiveView.UploadConfig{} = conf, opts \\ []) do
+    opts =
+      if conf.max_entries > 1 do
+        Keyword.put(opts, :multiple, true)
+      else
+        opts
+      end
+
+    assigns = %{conf: conf, drop_id: opts[:drop_target_id]}
+    ~L"""
+    <%= Phoenix.HTML.Tag.content_tag :input, "", Keyword.merge(opts,
+      type: "file",
+      id: @conf.ref,
+      name: @conf.name,
+      data_phx_upload_ref: conf.ref,
+      data_phx_active_refs: Enum.join(for(entry <- conf.entries, do: entry.ref), ","),
+      data_phx_drop_target_id: @drop_id) %>
```

Comment:
What happens here if `@drop_id` is not set? Should we only include `data_phx_drop_target_id` if it is explicitly specified in the opts?

Reply:
As it turns out this attribute is not necessary since the phx-drop-target references a DOM, which in this case will be the upload ref, ie 

```
<div class="container" phx-drop-target="<%= @uploads.avatar.ref %>">
   <%= live_file input @uploads.avatar %>
</div>
```

I will update the docs to include drop-target and kill this bit. Good catch!